libvast/src/system/importer.cpp

```diff
@@ -268,13 +269,8 @@ importer(importer_actor::stateful_pointer<importer_state> self,
   self->set_exit_handler([=](const caf::exit_msg& msg) {
     self->state.send_report();
     if (self->state.stage) {
-      self->state.stage->shutdown();
       self->state.stage->out().push(detail::framed<table_slice>::make_eof());
-      // We need to `fan_out_flush()` before we close, otherwise `close()`
-      // will delete all clean output paths and we might lose the eof.
-      self->state.stage->out().fan_out_flush();
-      self->state.stage->out().close();
-      self->state.stage->out().force_emit_batches();
+      detail::shutdown_stream_stage(self->state.stage);
```

Comment:
Does it significantly change the logic that we now effectively call shutdown after push rather than before it?

Reply:
As far as I can tell it doesn't: `shutdown()` only sends upstream messages but does not touch any of the path buffers.

Reply:
Thanks for double-checking. Fine by me then.